my-code/src/Components/MovieCard/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useNavigate } from 'react-router-dom';
+import { useMovieDetails } from '../../Contexts/MovieDetails';
+import { useMovies } from '../../Contexts/MoviesData';
+import { FavouriteIcon, MovieCardContainer, MovieDetails, MovieInfo, MoviePoster } from './styles';
+import fullHeartIcon from '/icons/icon-heart-full.svg';
+import whiteHeartIcon from '/icons/icon-heart-white.svg';
+
+interface MovieCardProps {
+    Poster: string
+    Title: string
+    Year: string
+    imdbID: string
```

Comment:
Is it possible if the api response is an object with capitalized properties? 
I know that in this case we dont have access to the api code, but wouldnt it be easier if this change was made in the api itself so the properties would be lowercase?

![image](https://user-images.githubusercontent.com/38993774/178811272-35f26f45-bd91-45e3-b7f5-88e72f511ab1.png)

Reply:
It is possible, since the only place you use `MovieCard` is on `MoviesList`. The refactor would be simple:
```
// Before
<MovieCard
	  key={movie.imdbID}
	  Poster={movie.Poster}
	  Title={movie.Title}
	  Year={movie.Year}
	  imdbID={movie.imdbID}
/>

// After
<MovieCard
	  key={movie.imdbID}
	  poster={movie.Poster}
	  title={movie.Title}
	  year={movie.Year}
	  imdbID={movie.imdbID}
/>
 ```
 
Now, if you were using it like this before:
```
<MovieCard
	  key={movie.imdbID}
	  {...movie}
/>
```
Things would be a bit more complicated. And I think we can come to a better, clearer solution:

Going to `Contexts/MoviesData`, you have an interface called `MovieProps`. This naming leads me to think that there is a component called `Movie`, and that these are its props. Since its not the case, instead, it could be called `Movie`, or `MovieData` or `MovieDatum`. Following this logic, `MoviesDataContextProps` could be `MoviesDataContext`/`MoviesDataContextType`/`MoviesDataContextT`, and `FavouriteMoviesProps`... well maybe it is not needed even:

```
// Before
const [favouriteMovies, setFavouriteMovies] = useState<FavouriteMoviesProps[]>
// After
const [favouriteMovies, setFavouriteMovies] = useState<Pick<Movie, 'imdbID'>>
```

This has the benefit of not having to create a new type, and it actually reflects the type of Movie. Meaning that if we change imdbID inside Movie to be, say, a number, due to API changes, we don't need to update in two places.

With that in mind, we can finally come to `MovieCard` again, and refactor it's type:
```
type MovieCardProps = Pick<Movie, "imdbID" | "Poster" | "Year" | "Title">;
```
With that same benefit of depending on the original type. But the props are still upper-case... In that case, I offer two alternatives:
```
type MovieCardProps = {
  imdbID: Movie['imdbID']
  title: Movie['Title']
  year: Movie['Year']
  poster: Movie['Poster']
}
// or
type MovieCardProps = {
  data: Movie;
}
```
That way the actual props are lower-case, while maintaining the Movie type as is, meaning only Movie accurately reflects the API. The component doesn't need to know what the API format is.